### ISSUE TITLE
Improve Error Handling and Resilience in LLM Service Integration

### ISSUE THREAD

**Issue by tawada** [tawada/grass-grower#49]
I've examined the provided program closely and would like to highlight an aspect that could potentially be improved for robustness and extensibility. This observation concerns the structure and usage of `services.llm` and its integration within the broader application. 

#### Description:
The integration of the `services.llm` module into the application, especially within `routers` and `logic` layers, presents an opportunity for enhancement in terms of error handling. While the current implementation allows the application to interact with an LLM service and perform various operations (like generating code from issues), it lacks comprehensive error handling mechanisms.

#### Example:
The `services.llm.generate_json` function can raise exceptions under certain conditions, such as API request failures. However, the invocations of this function in other parts of the application like in `logic.code_modification.generate_modification_from_issue` do not seem to adequately address potential failures of this function call. This can leave the application vulnerable to ungraceful failures and could disrupt the user experience.

```python
# In `services.llm.__init__.py`
def generate_json(...):
    ...
    raise RuntimeError("Failed to generate json after multiple retries")

# In `logic.code_modification.py`
def generate_modification_from_issue(...):
    ...
    generated_json = services.llm.generate_json(messages, openai_client)
    # ^ This can raise an uncaught RuntimeError.
```

#### Proposed Improvement:
Implement a more comprehensive and consistent error handling strategy across the application. This would involve catching potential exceptions where LLM services are consumed and either gracefully recovering or providing meaningful feedback to the user. Furthermore, extending the logging functionality to capture such error events could offer insights for debugging and enhancing system reliability.

In conclusion, while the application effectively incorporates LLM services for its functionalities, fortifying the error handling mechanisms can significantly bolster its resilience and user experience.
